core/common/src/test/java/alluxio/util/ShellUtilsTest.java

```diff
@@ -56,13 +56,25 @@ public void execCommandFail() throws Exception {
   }
 
   /**
-   * Tests the {@link ShellUtils#execCommand(String...)} method for a group of commands.
+   * Tests the {@link ShellUtils#execCommand(String...)} method for effective groups command.
    *
    * @throws Throwable when the execution of the commands fails
    */
   @Test
-  public void execGetGroupCommand() throws Exception {
-    String result = ShellUtils.execCommand(ShellUtils.getGroupsForUserCommand("root"));
+  public void execGetEffectiveGroupCommand() throws Exception {
+    String result = ShellUtils.execCommand(ShellUtils.getEffectiveGroupsForUserCommand("root"));
+    // On Linux user "root" will be a part of the group "root". On OSX it will be a part of "admin".
+    assertTrue(result.contains("root") || result.contains("admin"));
+  }
```

Comment:
Add additional test case such as :
```suggestion
  @Test
  public void execGetEffectiveGroupCommand() throws Exception {
    String maliciousName = "> echo ALLUXIO";
    ShellUtils.ExitCodeException exception =
        Assert.assertThrows(ShellUtils.ExitCodeException.class,
        () -> ShellUtils.execCommand(
        ShellUtils.getEffectiveGroupsForUserCommand(maliciousName))
    );
    Assert.assertTrue(exception.getMessage().contains("no such user"));
  }
```